File: app/src/main/java/com/wuruoye/know/base/IToolbarView.java
package com.wuruoye.know.base;

import android.support.annotation.DrawableRes;
import android.support.v7.widget.Toolbar;

public interface IToolbarView {
    interface OnToolbarBackListener {
        void onBackClick();
    }

    interface OnToolbarMoreListener {
        void onMoreClick();
    }

    interface OnToolbarTitleListener {
        void onTitleClick();
    }

    Toolbar getToolbar();
    void setToolbarBackListener(OnToolbarBackListener listener);
    void setToolbarMoreListener(OnToolbarMoreListener listener);
    void setToolbarTitleListener(OnToolbarTitleListener listener);
    void setToolbarBack(@DrawableRes int resource, String title);
    void setToolbarMore(@DrawableRes int resource, String title);
    void setToolbarTitle(String title);
}

File: app/src/main/java/com/wuruoye/know/base/IToolbarView.kt
package com.wuruoye.know.base

import android.support.annotation.DrawableRes
import android.support.v7.widget.Toolbar

interface IToolbarView {

    interface OnToolbarBackListener {
        fun onBackClick()
    }

    interface OnToolbarMoreListener {
        fun onMoreClick()
    }

    interface OnToolbarTitleListener {
        fun onTitleClick()
    }

    fun setToolbarBackListener(listener: OnToolbarBackListener)
    fun setToolbarMoreListener(listener: OnToolbarMoreListener)
    fun setToolbarTitleListener(listener: OnToolbarTitleListener)
    fun setToolbarBack(@DrawableRes resource: Int, title: String)
    fun setToolbarMore(@DrawableRes resource: Int, title: String)
    fun setToolbarTitle(title: String)
    fun getToolbar(): Toolbar
}
